.pipelines/ci.yml

```diff
@@ -41,19 +37,56 @@ variables:
     value: "!smoke&&!regressiontest"
 
 stages:
+  - stage: Set_Tag_Stage
+    displayName: "Set Dynamic TAG Variables"
+    jobs:
+      - job: Set_Tag_Variables
+        displayName: "Set TAG Variables"
+        pool:
+          name: 1es-aro-ci-pool
+        steps:
+          - checkout: self
+
+          - script: |
+              echo "Setting dynamic variables..."
```

Comment:
Can we not just have this stage output a single `TAG` variable? Do we care in any downstream steps if it's a PR vs branch?

Reply:
Thanks for the suggestion, Tanmay! I've simplified the logic to output a single TAG variable, removing the need for separate PR_TAG and MASTER_TAG. The downstream steps now use this unified TAG directly, ensuring consistent tagging across all stages.